lib/bitcoin/lock/LockMonitor.ts

```diff
@@ -0,0 +1,304 @@
+import BitcoinClient from '../BitcoinClient';
+import BitcoinError from '../BitcoinError';
+import BitcoinLockTransactionModel from '../models/BitcoinLockTransactionModel';
+import ErrorCode from '../ErrorCode';
+import LockIdentifier from '../models/LockIdentifierModel';
+import LockIdentifierSerializer from './LockIdentifierSerializer';
+import LockResolver from './LockResolver';
+import LockTransactionModel from './../models/LockTransactionModel';
+import LockTransactionType from './../enums/LockTransactionType';
+import MongoDbLockTransactionStore from './MongoDbLockTransactionStore';
+import ValueTimeLockModel from './../../common/models/ValueTimeLockModel';
+
+/**
+ * Structure (internal to this class) to track the information about lock.
+ */
+interface LockInformation {
+  currentValueTimeLock: ValueTimeLockModel | undefined;
+  latestSavedLockInfo: LockTransactionModel | undefined;
+
+  // Need to add a 'state'
+}
+
+/**
+ * Encapsulates functionality to monitor and create/remove amount locks on bitcoin.
+ */
+export default class LockMonitor {
+
+  private periodicPollTimeoutId: number | undefined;
+
+  private currentValueTimeLock: LockInformation | undefined;
+
+  private lockResolver: LockResolver;
+
+  constructor (
+    private bitcoinClient: BitcoinClient,
+    private lockTransactionStore: MongoDbLockTransactionStore,
+    private pollPeriodInSeconds: number,
+    private desiredLockAmountInSatoshis: number,
+    private lockPeriodInBlocks: number,
+    private firstLockFeeAmountInSatoshis: number) {
+
+    this.lockResolver = new LockResolver(this.bitcoinClient);
+  }
+
+  /**
+   * Initializes this object by performing the periodic poll tasks.
+   */
+  public async initialize (): Promise<void> {
+    await this.periodicPoll();
+  }
+
+  private async periodicPoll (intervalInSeconds: number = this.pollPeriodInSeconds): Promise<void> {
+
+    try {
+      // Defensive programming to prevent multiple polling loops even if this method is externally called multiple times.
+      if (this.periodicPollTimeoutId) {
+        clearTimeout(this.periodicPollTimeoutId);
+      }
+
+      console.info(`Starting periodic polling for the lock monitor.`);
+      await this.handlePeriodicPolling();
+
+    } catch (e) {
+      const message = `An error occured during periodic poll: ${JSON.stringify(e, Object.getOwnPropertyNames(e))}`;
+      console.error(message);
+    } finally {
+      this.periodicPollTimeoutId = setTimeout(this.periodicPoll.bind(this), 1000 * intervalInSeconds, intervalInSeconds);

```

Comment:
Following up to Isaac's comment above.
```suggestion
      this.periodicPollTimeoutId = setTimeout(this.periodicPoll.bind(this), 1000 * intervalInSeconds);

```